plugins/copr.py

```diff
@@ -456,8 +462,12 @@ def _guess_chroot(self):
     def _download_repo(self, project_name, repo_filename, chroot=None):
         if chroot is None:
             chroot = self._guess_chroot()
-        short_chroot = '-'.join(chroot.split('-')[:2])
-        arch = chroot.split('-')[2]
+        if len(chroot.split("-")) == 4:
+            short_chroot = '-'.join(chroot.split('-')[:3])
+            arch = chroot.split('-')[3]
+        else:
+            short_chroot = '-'.join(chroot.split('-')[:2])
+            arch = chroot.split('-')[2]
```

Comment:
What if the length of the splitted list is 5 or greater? Will this do what you expect? Maybe the initial check on line 229 should be for length equal to 3 or 4?

You are also calling `chroot.split()` multiple times, while in practice it probably has no impact, it's not nice, you should split once, store the splitted list and use that.

Reply:
if the list is longer, it still works. Currently 2 is taken as the index on line 465, but this obviously doesn't matter, because for example api_path /coprs/schlupov/test_chroot/repo/opensuse-leap-15.2/dnf.repo?arch=x86_64 for some reason works the same as /coprs/schlupov/test_chroot/repo/opensuse-leap/dnf.repo?arch=15.2.

Reply:
Well, I don't find anything too obvious about this :/

The fact the two URLs both work is already quite confusing and just coincidental, I guess it's because the last part of the path is joined with the `arch` argument using a `"-"`, but I wonder what the architecture is for the latter URL. Is there a default of `"x86_64"` in case the architecture is missing?

So for e.g. a chroot like this: `"a-b-c-d-arch"` your current code will process it so that `short_chroot = "a-b"`, `arch = "c"` and `"d-arch"` will just be ignored, right? I haven't tried to run it to see what the error message would be ultimately, but I would guess it can't be right. Since I assume there is no valid chroot that has five elements like this, you may think it's not an issue, but if a user enters such chroot just by mistake, the error message they'll get will just be confusing.

You are imposing a structure by "parsing" the chroot and you should handle all corner cases.

You can just check the chroot splitted length is either 3 or 4 in the check on line 229 and actually also include length of 4 variant in the error message (right now it says the format is `"distribution-version-architecture"` but for length of 4 that's different.

Reply:
> just coincidental

That's not a coincidence, it's the intentional design of that route (you can name the '*.repo' link as you wish).

> You can just check the chroot splitted length is either 3 or 4

There's no reason to limit that to 3/4.  And actually, this is a can of worms, because e.g. mock-rpmfusion-free
package doesn't hold the `arch` on last position and it has more parts.  So strictly speaking, the plugin needs
some major re-design and this PR isn't the right time/place for it.

We should concentrate on the traceback.